File: array/118.java
import java.util.*;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ret = new ArrayList<>();
        for (int index = 1; index <= numRows; index++) {
            List<Integer> curList = new ArrayList<>();
            curList.add(1);
            if (index == 1) {
                ret.add(curList);
                continue;
            }
            if (index == 2) {
                curList.add(1);
                ret.add(curList);
                continue;
            }
            for (int i = 1; i < index-1; i++) {
                curList.add(ret.get(index-2).get(i-1) + ret.get(index-2).get(i));
            }
            curList.add(1);
            ret.add(curList);
        }
        return ret;
    }
}

File: array/118.cc
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        std::vector<std::vector<int>> ret;
        ret.emplace_back(std::vector<int>(1,1));
        for (int i = 1; i < numRows; ++i) {
            std::vector<int> cur(1,1);
            for (int j = 0; j < (i-1); ++j) {
                cur.emplace_back(ret[i-1][j] + ret[i-1][j+1]);
            }
            cur.emplace_back(1);
            ret.emplace_back(std::move(cur));
        }
        return ret;
    }
};
